docs/build-css-js.md

```diff
@@ -0,0 +1,51 @@
+# Building CSS and JS
+
+This example uses `npm` and [Webpack Encore](https://symfony.com/doc/current/frontend.html).
+
+Modify `Dockerfile` to build your assets before anything else.
+The output in `public/build` (the configuration default) will then be copied to both caddy and php.
+
+In your production environment, this is all you need for an immutable release.
+
+```diff
+ARG CADDY_VERSION=2
++ ARG NODE_VERSION=17
+
++ FROM node:${NODE_VERSION}-alpine AS symfony_npm
++ WORKDIR /srv/app
++ COPY . .
++ RUN npm install; \
++     npm run build
+
+# "php" stage
+```
+
+```diff
+VOLUME /srv/app/var
+
++ COPY --from=symfony_npm /srv/app/public/build public/build
+
+ENTRYPOINT ["docker-entrypoint"]
+CMD ["php-fpm"]
+
+FROM caddy:${CADDY_VERSION}-builder-alpine AS symfony_caddy_builder
+```
+
+Modify `docker-compose.override.yml` to add an `npm` container in your development environment and watch for file changes.
+
+```diff
+  caddy:
+    volumes:
+      - ./docker/caddy/Caddyfile:/etc/caddy/Caddyfile:ro
+      - ./public:/srv/app/public:ro
+
++  npm:
++    build:
++      context: .
++      target: symfony_npm
++    depends_on:
++      - caddy
++    volumes:
++      - ./:/srv/app:rw,cached
++    command: 'sh -c "npm install; npm run watch"'
```

Comment:
Wouldn't it be better to use the [dev server](https://symfony.com/doc/current/frontend/encore/dev-server.html) instead?

Reply:
If you want to use hot module reloading, then yes. I think we have to expose some other ports as well. I will investigate it further.

Reply:
@dunglas 

After investigating a bit further I've hit a stumbling block.

Out of the box, webpack-dev-server will use http only over port 8080. The default dev-server host option is localhost, which means you can't access it from your host machine.

If we use

    command: 'sh -c "npm install; npm run dev-server -- --https --host 0.0.0.0 --allowed-hosts all"'

Then the container is accessible (e.g. you can load https://localhost:8080/build/app.css) but the URLs generated will be https://0.0.0.0:8080/build/app.css, which don't work.

I can't see anything in the docs that show how to change the host that it binds too AND also set the host that is generated in URLs.

I think I did try getting the dev-server running a while ago and ran into this same problem. I personally don't use HMR or live-reloading, so the npm run watch command is usually fine for me.

Reply:
in my case it is enough run `npm run dev-server`. And i share the ssl cart from caddy. 

https://symfony.com/doc/current/frontend/encore/dev-server.html#dev-server-options
```js
// webpack.config.js
// ...

Encore
    // ...

    .configureDevServerOptions(options => {
        options.https = {
            key: '/path/to/server.key',
            cert: '/path/to/server.crt',
        }
    })
;
```

Reply:
> In my case it is enough run `npm run dev-server`. And i share the ssl cert from caddy.
> 
> https://symfony.com/doc/current/frontend/encore/dev-server.html#dev-server-options
> 
> ```js
> // webpack.config.js
> // ...
> 
> Encore
>     // ...
> 
>     .configureDevServerOptions(options => {
>         options.https = {
>             key: '/path/to/server.key',
>             cert: '/path/to/server.crt',
>         }
>     })
> ;
> ```

Are you doing this from your host OS though? I'm trying to get it running within the node container  - so there is no dependency on having node running on your local machine.